dbms/src/Storages/DeltaMerge/ReadThread/SegmentReader.cpp

```diff
@@ -116,6 +109,11 @@ class SegmentReader
             {
                 LOG_FMT_DEBUG(log, "All finished, pool_ids={} segment_id={} read_count={}", merged_task->getPoolIds(), merged_task->getSegmentId(), read_count);
             }
+
+            if (!merged_task->allStreamsFinished())
```

Comment:
Add some comments that why we should not re-push the merged task if exception thrown in the previous lines

Reply:
And can you remind the reason why you use `SCOPE_EXIT` to re-push the merged task?